src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -65,13 +65,15 @@ def _cancel_timer_event_handler(_, **__):
 
 
 # region PARAMETER VALIDATORS
-def parse_storage_account(namespace):
+def parse_storage_account(cmd, namespace):
     """Parse storage account which can be either account name or account id"""
     from msrestazure.tools import parse_resource_id, is_valid_resource_id
 
     if namespace.account_name and is_valid_resource_id(namespace.account_name):
         namespace.resource_group_name = parse_resource_id(namespace.account_name)['resource_group']
         namespace.account_name = parse_resource_id(namespace.account_name)['name']
+    elif not namespace.resource_group_name:
```

Comment:
what's the behavior if both account_name and resource_group_name are None?

Reply:
Good catch! I will add one more condition for the function although account-name is required for all commands. 😊